stl/src/xlexp.cpp

```diff
@@ -18,11 +18,14 @@ static const long double invln2 = 1.4426950408889634073599246810018921L;
 
 _CRTIMP2_PURE short __CLRCALL_PURE_OR_CDECL _LExp(
     long double* px, long double y, short eoff) { // compute y * e^(*px), (*px) finite, |y| not huge
-    if (*px < -hugexp || y == 0.0L) { // certain underflow
-        *px = 0.0L;
+    if (y == 0.0L) { // zero
+        *px = y;
+        return 0;
+    } else if (*px < -hugexp) { // certain underflow
+        *px = 0.0L * y;
         return 0;
     } else if (hugexp < *px) { // certain overflow
-        *px = _LInf._Long_double;
+        *px = _LInf._Long_double * y;

```

Comment:
Similarly, `xexp.cpp` said `*px = _Xfe_overflow(y);` - should this be consistent?